### ISSUE TITLE
flashing "arduino_101_sss" build onto Arduino 101 breaks DFU

### ISSUE THREAD

**Issue by nashif** [zephyriot/zep-jira14#751]
**_Reported by Andrew Boie:_**

We have two arduino_101 and arduino_101_sss targets, one for custom bootloader and one for factory bootloader with DFU.

In the custom configuration, the base flash addresses are:

arduino_101:  0x40030000
arduino_101_sss: 0x40000000

In the factory configuration:

arduino_101_factory: 0x40010000
arduino_101_sss_factory: 0x40034000

Note the 64K gap between 0x40000000 and 0x40010000. When single-stepping from boot with the factory bootloader, the device makes a jump to 0x40000030. This is presumably a 2nd-stage bootloader which implements DFU.

If someone flashes the arduino_101_sss build onto a device, whatever code was in that region gets blow away in favor of the ARC Zephyr image, and the device fails to boot until the custom bootloader from QMSI is used, and DFU is not possible.

(Imported from Jira ZEP-844)

**Comment by nashif** [comment]
**_by Andrew Boie:_**

This should fix it
https://gerrit.zephyrproject.org/r/#/c/4632

**Comment by nashif** [comment]
**_by Qiu Peiyang:_**

I did some search and suspect https://gerrit.zephyrproject.org/r/#/c/4687/ is the fix for GH-751.
According to the description of this bug, before this patch is applied, Arduino_101 board will fail to boot because the second stage bootloader between 0x40000000 ~ 0x40010000 is overwritten by ARC Zephyr image.
So the idea is, if we can get Arduino_101 boot up successfully after flashing ARC cases using DFU, then this patch can be verified work, and this bug can be closed.

I just tested it, and “Hello World” on ARC and X86 can both be printed correctly. So close this bug.
